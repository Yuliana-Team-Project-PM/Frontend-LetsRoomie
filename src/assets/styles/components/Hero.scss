@import '../Vars.scss';

.Hero {
    height: 55vh;
    background-image: url('https://images.unsplash.com/photo-1600493505371-f2f6153dbb29?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80');
    background-repeat: no-repeat; 
    background-position: center;
    background-size: 100vw;
    position: relative;
    

    &__gradient {
        background: linear-gradient(90deg, rgba(255,255,255,.80) 40%, rgba(255,255,255,0) 100%);
        height: 55vh;
    }

    &__title {
        position: absolute;
        top: 40%;
        left: 18%;
        width: 30%;

        &--description {
            color: $primary-color;
            margin: 0;
            font-weight: normal;

            & span {
                font-weight: bold;
            }
        }

        &--button {
            @include base-button ($gray-color, $secondary-color, $border-button);
            font-size: 1em;
            margin: .7em 0;
        }
    }
}

@media screen and (max-width: 768px) {
    .Hero {
        &__title {
            top: 30%;
            & h1 {
                font-size: 1.5em;
            }
            &--button {
                font-size: .8em;
            }
        }
    }
}

@media screen and (max-width: 425px) {
    .Hero {
        height: 36vh;

        &__gradient {
            height: 36vh;
        }

        &__title {
            & h1 {
                font-size: 1em;
            }
            &--button {
                font-size: .6em;
                padding: .6em;
                border-radius: 5px;
            }
        }
    }
}

@media screen and (max-width: 426px) {
    .Hero {
        height: 39vh;
        
        &__gradient {
            height: 39vh;
        }

        &__title {
            top:17%;

            & h1 {
                font-size: .9em;
            }
            &--button {
                font-size: .5em;
                padding: .5em;
                border-radius: 5px;
            }
        }
    }
}

@media screen and (max-width: 375px) {
    .Hero {
        height: 27vh;

        &__gradient {
            height: 27vh;
        }
    }
}

@media screen and (max-width: 375px) {
    .Hero {
        &__title {
            top: 23%;

            & h1 {
                font-size: .8em;
            }
        }
    }
}

