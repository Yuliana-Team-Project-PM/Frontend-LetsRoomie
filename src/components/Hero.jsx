import React from 'react';
import { Link } from 'react-router-dom';

import '../assets/styles/components/Hero.scss';

const Hero = () => {
    return(
        <section className='Hero'>
            <div className='Hero__gradient'></div>
            <div className='Hero__title'>
                <h1 className='Hero__title--description'>Te ayudamos a encontrar tu <span>espacio ideal</span> con el <span>ROOMIE</span> correcto.</h1>
                <Link to='select-acount'>
                    <button className='Hero__title--button'>Ãšnete</button>
                </Link>
            </div>
        </section>
    );
};

export default Hero;