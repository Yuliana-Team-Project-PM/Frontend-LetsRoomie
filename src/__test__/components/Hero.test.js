import React from 'react';
import { shallow } from 'enzyme';
import Hero from '../../components/Hero';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';


describe('<Hero />', () => {
    const hero = shallow(<Hero />);
    // const hero = mount(
    //     <MemoryRouter>
    //         <Hero />
    //     </MemoryRouter>
    // );

    test('Render de componente Hero', () => {
        expect(hero.length).toEqual(1);
    });

    test('Hero contiene un Link', () => {
        expect(hero.find('Link')).toHaveLength(1);
    });

    test('Hero contiene un botón', () => {
        expect(hero.find('button')).toHaveLength(1);
    });

    test('Hero, título correcto', () => {
        expect(hero.find('h1').text()).toEqual('Te ayudamos a encontrar tu espacio ideal con el ROOMIE correcto.');
    });

    test('Hero Snapshot', () => {
        const hero = create(
            <MemoryRouter>
                <Hero />
            </MemoryRouter>
        );
        expect(hero.toJSON()).toMatchSnapshot();
    })
})
