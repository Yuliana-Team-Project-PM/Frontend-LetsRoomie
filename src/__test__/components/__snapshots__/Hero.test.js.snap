// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Hero Snapshot 1`] = `
<section
  className="Hero"
>
  <div
    className="Hero__gradient"
  />
  <div
    className="Hero__title"
  >
    <h1
      className="Hero__title--description"
    >
      Te ayudamos a encontrar tu 
      <span>
        espacio ideal
      </span>
       con el 
      <span>
        ROOMIE
      </span>
       correcto.
    </h1>
    <a
      href="/select-acount"
      onClick={[Function]}
    >
      <button
        className="Hero__title--button"
      >
        Ãšnete
      </button>
    </a>
  </div>
</section>
`;
